Branch-SDK/src/io/branch/referral/DeviceInfo.java

```diff
@@ -105,6 +106,10 @@ void updateRequestWithV1Params(JSONObject requestObj) {
                 requestObj.put(Defines.Jsonkey.LocalIP.getKey(), localIpAddr);
             }
 
+            String imei = SystemObserver.getImei(context_);
```

Comment:
Question:  Isn't this expensive?   Except for the very very few customers where this might be used, every single request has to go down the path of calling `getSecondaryRequestMetaData`, and doing a JSON lookup for a string... and checking to see if that succeeded -- all of which most of the time will fail.

It's all on a background thread but ... again, seems expensive.